### DISCUSSION TITLE
Remote server for git operations

### DISCUSSION THREAD

**Discussion by jacobmichels** [Vinzent03/obsidian-git#831] [General]
👋 love the plugin 😄 

The only real issue I've encountered with this plugin is mobile support (iOS in my case). It's unfortunate that the plugin is limited by Obsidian's sandbox and isomorphic git's inefficiency. Wondering if it was considered to use a remote server for git operations.

The way I envision it looking is
- user hosts a specialized webserver application (that we'd build) on a server they own. this application would do two jobs
  - perform git operations on the user's vault (stored locally) when the client requests 
  - perform vault modifications on the user's vault as dictated by client requests (ie. user changes vault locally, change is streamed to server, server applies same change to keep vaults consistent)
- the obsidian plugin would attempt to connect to this server if defined in the plugin settings
  - we could fall back to isomorphic shit if we couldn't establish the connection
- any changes made to files within the vault are streamed to the server
- any git operations are performed on the server, results streamed to the client
- the result _ideally_ is what appears to be using git locally on the client

Some concerns
- the capabilities the webserver provides are extremely powerful; it could be abused if a malicious actor gained access to it. 
- streaming files, could be slow/data hungry if many changes are made
- git operation ordering
  - user adds a massive file to their vault, the git plugin begins to stream it the file the remote git host async
  - user requests a commit-and-sync operation
  - does the remote server wait for the file stream to complete?
    - if a new stream is initiated while waiting for the existing one, does the git host also wait for that one?

Curious to hear others' thoughts on this approach.

**Comment by Vinzent03** [comment]
Thanks for your enthusiasm, but this won't ever make it into this plugin. Especially on mobile with weak internet connection, this is not doable. I much rather suggest to use other programs on mobile to do the git work than live syncing your whole vault with some server.
